### PR TITLE
Allow a column other than `id` to represent the cursor.

### PR DESCRIPTION
**Pull request by thaterikperson**
If you're using `Ecto.UUID` as your primary key, there is no guarantee that two objects' ids will be ordered  alphanumerically.

For instance, given a set of objects that looks like this:

``` elixir
a = %{id: "e0000000-0000-0000-0000-000000000000",
      inserted_at: #Ecto.DateTime<2016-06-10T00:00:00Z>}
b = %{id: "e0000000-0000-0000-0000-000000000001",
      inserted_at: #Ecto.DateTime<2016-06-11T00:00:00Z>}
c = %{id: "d0000000-0000-0000-0000-000000000000",
      inserted_at: #Ecto.DateTime<2016-06-12T00:00:00Z>}
```

Calling

``` elixir
Connection.Ecto.resolve(Repo, query, %{after: a_cursor})
```

Results in

``` elixir
%{
  edges: [
    %{node: b, cursor: b_cursor}
  ],
  pageInfo: %{
    startCursor: b_cursor,
    endCursor: b_cursor,
    hasPreviousPage: false,
    hasNextPage: false,
  }
}
```

The `c` object is not included in the results because `a.id > c.id` and the code currently assumes `id` will be an auto-incrementing integer.
### Solution

Allow user to pass in an arbitrary property that will be used to paginate the results.

``` elixir
Connection.Ecto.resolve(Repo, query, %{after: @a_cursor, where: :inserted_at})
```

Results in

``` elixir
%{
  edges: [
    %{node: b, cursor: b_cursor},
    %{node: c, cursor: c_cursor}
  ],
  pageInfo: %{
    startCursor: b_cursor,
    endCursor: c_cursor,
    hasPreviousPage: false,
    hasNextPage: false,
  }
}
```

### CODE REVIEW

**Comment by thaterikperson** [comment]
The `sqlite_ecto` driver doesn't support the way I'm using fragments, so tests will fail until #23 is merged.

**Comment by seanabrahams** [comment]
@thaterikperson Thanks for the PR! I don't have time right now to review but I have merged in #23 if you want to rebase. Thanks again and I'll check back into this later. Cheers!

File: lib/graphql/relay/connection/ecto.ex [hunk 1]
```diff
@@ -63,16 +63,17 @@ if Code.ensure_loaded?(Ecto) do
       a_after = cursor_to_offset(args[:after])
       first = args[:first]
       last = args[:last]
+      where_property = args[:where] || :id
```
Comment (seanabrahams) [inline lib/graphql/relay/connection/ecto.ex]: @thaterikperson Thanks again for the PR and attention to detail including cleaning up the tests. Appreciate it!

I'm still grokking this and wondering if the argument should be named something other than `where` as I'm finding it a bit confusing. Maybe `order_by`, `ordered_by`, `sort_by`, or `sorted_by`?

``` elixir
Connection.Ecto.resolve(Repo, query, %{after: @a_cursor, ordered_by: :inserted_at})
```

What do you think?

Comment (thaterikperson) [inline lib/graphql/relay/connection/ecto.ex]: I like `ordered_by` as well. Think that's a little more intuitive. Pushed a fix.

**Comment by seanabrahams** [comment]
I've created a new branch+PR that includes this PR plus the ability to set the direction the field is sorted by. See https://github.com/graphql-elixir/graphql_relay/pull/26 Thanks @thaterikperson for the original PR!!! 👍 👍 👍 😄 😄 😄

**Pull request closed without merging**